bcgov/cas-registration
# feat:Activity-Hydrogen Production

Card: https://github.com/bcgov/cas-reporting/issues/289. This adds the Hydrogen production activity.

**Changes:-**
- Adds backend config/migration for Hydrogen production activity.
- Adds frontend page and components for Hydrogen production activity.
- Added custom_methodology_schema model for handling complex methodology logic.
- Updated build_form_schema to apply custom schemas dynamically.
- Added any missing configurations and relationships to the database.Created routing for industry_user and industry_user_admin flows.
- Added components for activity-specific pages and dynamic field rendering.

**To test:-**
- make reset_db to apply latest migrations.
- spin up the front, backend and postgres.
- Login, then navigate to http://localhost:3000/reporting/reports/1/facilities/1/activities/hydrogemr_production, click through the available form fields.
- The Gas type dropdown options should change options available in the methodology field.
- The Methodology dropdown options spawn additional fields, depending on the option selected.
